pkg/elemental/elemental.go

```diff
@@ -549,6 +549,11 @@ func DeployRecoverySystem(cfg types.Config, img *types.Image, bootDir string) er
 		"linux":   kernel,
 		"initrd":  initrd,
 	} {
+		// if kernel/initrd is actually named "vmlinuz"/"initrd" we just skip the symlink part.
+		if kernel == name || initrd == name {
+			continue
```

Comment:
It might be a good idea to add a `Debugf` call just to simplify the debugging in case of issue:
```suggestion
			cfg.Logger.Debugf("No symlink needed, keeping exising %s file", name)
			continue
```